test/features/thread/presentation/widgets/email_tile_web_builder_test.dart

```diff
@@ -0,0 +1,206 @@
+import 'package:core/presentation/extensions/color_extension.dart';
+import 'package:core/presentation/resources/image_paths.dart';
+import 'package:core/presentation/utils/responsive_utils.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_localizations/flutter_localizations.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:get/get.dart';
+import 'package:jmap_dart_client/jmap/mail/email/keyword_identifier.dart';
+import 'package:model/email/presentation_email.dart';
+import 'package:model/mailbox/select_mode.dart';
+import 'package:tmail_ui_user/features/thread/presentation/widgets/email_tile_web_builder.dart';
+import 'package:tmail_ui_user/main/localizations/app_localizations_delegate.dart';
+import 'package:tmail_ui_user/main/localizations/localization_service.dart';
+
+void main() {
+  setUpAll(() {
+    Get.put(ResponsiveUtils());
+    Get.put(ImagePaths());
+  });
+
+  group('EmailTileBuilder test:', () {
+    testWidgets(
+      'Should display checkbox image '
+      'and InkWell with hover color AppColor.colorEmailTileHoverWeb '
+      'when app is running in desktop layout',
+    (tester) async {
+      // arrange
+      final widgetUnderTest = EmailTileBuilder(
+        presentationEmail: PresentationEmail(),
+        selectAllMode: SelectMode.INACTIVE,
+        isShowingEmailContent: false,
+      );
+      debugDefaultTargetPlatformOverride = TargetPlatform.macOS;
+      final dpi = tester.view.devicePixelRatio;
+      tester.view.physicalSize = Size(dpi * 1920 * 2, dpi * 1080 * 2);
+
+      // act
+      await tester.pumpWidget(
+        GetMaterialApp(
+          localizationsDelegates: const [
+            AppLocalizationsDelegate(),
+            GlobalMaterialLocalizations.delegate,
+            GlobalWidgetsLocalizations.delegate,
+            GlobalCupertinoLocalizations.delegate],
+          supportedLocales: LocalizationService.supportedLocales,
+          home: Scaffold(body: widgetUnderTest)));
+      await tester.pump();
+
+      // assert
+      expect(
+        find.byWidgetPredicate(
+          (widget) => widget is SvgPicture 
+            && widget.bytesLoader is SvgAssetLoader
+            && (widget.bytesLoader as SvgAssetLoader).assetName == ImagePaths().icCheckboxUnselected
+        ),
+        findsOneWidget);
+      expect(
+        find.byWidgetPredicate(
+          (widget) => widget is InkWell
+            && widget.hoverColor == AppColor.colorEmailTileHoverWeb
+        ),
+        findsOneWidget);
+
+      // clean up
+      debugDefaultTargetPlatformOverride = null;
+      tester.view.reset();
+    });
+
+    testWidgets(
+      'Should not display any checkbox image '
```

Comment:
IMO, we always show check box. why we have this test?

Reply:
ah, not running in desktop layout. why we know this test not run in desktop layout? it is by default?

Reply:
The default dimension for widget test is 360 x 600